src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -216,6 +218,11 @@ private SynthesizedBackingFieldSymbol CreateBackingField(bool isCreatedForFieldK
                 Interlocked.CompareExchange(ref _lazyBackingFieldSymbol, backingField, _lazyBackingFieldSymbolSentinel);
             }
 
+            var type = this.Type;
+            if (type.IsRefLikeType && (this.IsStatic || !this.ContainingType.IsRefLikeType))
+            {
+                _diagnostics.Add(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, TypeLocation, type);
+            }
```

Comment:
Moved this from `ValidatePropertyType` so that I don't need to check whether this is an auto or semi auto property or a regular property.